src/main/java/net/tammon/sip/TCPConnection.java

```diff
@@ -39,465 +39,578 @@
-	 */
-	public Data readData(int slaveIndex, int slaveExtension, String idn) throws SipException {
-		ReadOnlyData request = new ReadOnlyData(this.getNewTransactionId(), (short) slaveIndex, (short) slaveExtension,
-				idn);
-		ReadOnlyDataResponse response = (ReadOnlyDataResponse) this.getTcpResponse(request, ReadOnlyDataResponse.class);
-		
-		return response.getData();
-	}
-
-	/**
-	 * Checks whether or not the TCP connection to the sercos slave is still
-	 * connected
-	 *
-	 * @return true if the connection is alive
-	 */
-	@Override
-	public boolean isConnected() {
-		synchronized (this) {
-			return connected && this.socketConnection.isConnected();
-		}
-	}
-
-	/**
-	 * Returns a list of the supported message types of the sercos device. This list
-	 * is initially sent by the device during the sip connection process.
-	 *
-	 * @return the list of supported message types
-	 */
-	@Override
-	public List<Integer> getSupportedMessages() {
-		synchronized (this) {
-			return supportedMessages;
-		}
-	}
-
-	/**
-	 * Returns the IP address of the sercos device which is used by this TCP
-	 * connection.
-	 *
-	 * @return IP address of sercos device as {@link InetAddress}
-	 */
-	@Override
-	public InetAddress getIpAddress() {
-		return ipAddress;
-	}
-
-	/**
-	 * Returns the used sercos device port of the current sip connection
-	 *
-	 * @return used sercos device port
-	 */
-	@Override
-	public int getSipPort() {
-		return sipPort;
-	}
-
-	/**
-	 * Returns the used version of the Sercos Internet Protocol
-	 *
-	 * @return version of Sercos Internet Protocol
-	 */
-	@Override
-	public int getSipVersion() {
-		return sipVersion;
-	}
-
-	private void restartKeepAliveTimer() {
-		if (Objects.isNull(this.executorService))
-			this.executorService = Executors.newScheduledThreadPool(1);
-		else
-			this.executorService.shutdownNow();
-		final long period = Math.round(this.leaseTimeout * 0.7);
-		executorService.scheduleAtFixedRate(this::respondsToPing, 0, period, TimeUnit.MILLISECONDS);
-	}
-
-	/**
-	 * Stops the keep alive loop and closes the socket connection to the sercos
-	 * device
-	 */
-	@Override
-	public void disconnect() {
-	  
-	  close();
-
-		if (executorService != null) {
-			try {
-				executorService.shutdownNow();
-			} catch (Exception e) {
-				// No exception handling here
-			}
-		}
-
-		try {
-		  synchronized (this) {
-			  if (socketConnection != null) {
-				  socketConnection.close();
-				  socketConnection = null;
-			  }
-		  }
-		} catch (IOException e) {
-			e.printStackTrace();
-		}
-	}
-
-  private void close() {
-    try {
-      if (dataInputStream != null) {
-        dataInputStream.close();
-        dataInputStream = null;
-      }
-      if (dataOutputStream != null) {
-        dataOutputStream.close();
-        dataOutputStream = null;
-      }
-    } catch (Throwable e) {
-      e.printStackTrace();
+    
+    private static Logger _logger = LoggerFactory.getLogger("net.tammon.sip");
```

Comment:
Is there a specific reason why the logger shouldn't use the class name? This would possibly remove information
```suggestion
    private static Logger _logger = LoggerFactory.getLogger(TCPConnection.class);
```